internal/gatewayapi/contexts.go

```diff
@@ -330,6 +304,7 @@ type RouteParentContext struct {
 
 	routeParentStatusIdx int
 	listeners            []*ListenerContext
+	gateway              *GatewayContext
```

Comment:
isnt the `gateway` context part of the listener context ?

Reply:
the `gateway` of listener context is a `*gwapiv1.Gateway`. we need the `envoyproxy` in the gateway context.

we can also change the the `gateway` to a gateway context in listener context and reference it via `RouteParentContext.listeners[0].gateway`.

Reply:
yeah moving listenContext field to `GatewayContext` would be better, imo the code wouldnt need to change since `*gwapiv1.Gateway` is a embedded field in `GatewayContext`